### ISSUE TITLE
uuid of player doesn't work

### ISSUE THREAD

**Issue by MuratKandilli** [SkriptLang/Skript#4821]
### Skript/Server Version

```markdown
[13:01:09 INFO]: [Skript] Skript's aliases can be found here: https://github.com/SkriptLang/skript-aliases
[13:01:09 INFO]: [Skript] Skript's documentation can be found here: https://skriptlang.github.io/Skript
[13:01:09 INFO]: [Skript] Server Version: git-Paper-379 (MC: 1.18.2)
[13:01:09 INFO]: [Skript] Skript Version: 2.6.2
[13:01:09 INFO]: [Skript] Installed Skript Addons: None
[13:01:09 INFO]: [Skript] Installed dependencies: None
```

### Bug Description

When I try to use "uuid of player" expression it doesn't work. İt gives me error like this;

[13:00:38 INFO]: Line 3: (test.sk)
[13:00:38 INFO]:     Can't understand this expression: 'uuid of player'
[13:00:38 INFO]:     Line: set {_uuid} to uuid of player

I tried everything but I cant do anything

### Steps to Reproduce

command /givemyuuid:
    trigger:
        set {_uuid} to uuid of player
        broadcast "%{_uuid}%"

### Errors or Screenshots

![image](https://user-images.githubusercontent.com/25253555/174475763-36f7f77a-f805-4d03-bcb9-9e9ada514512.png)

### Agreement

- [X] I have read the guidelines above and affirm I am following them with this report.

**Comment by sovdeeth** [comment]
When they say "tried everything", they've tried re-installing both skript and the whole server, tried updating to java 17, tried with and without addons, even tried `player's uuid` just to be sure. No one else was able to replicate so I told them to make an issue here hoping that people with more technical knowledge might have an idea about what's going on.
Conversation occurred in the SkUnity skript-help-2 channel, here's [one link](https://discord.com/channels/135877399391764480/138027100593455104/988015075623338065) but the conversation was spread out.

**Comment by AyhamAl-Ali** [comment]
Can you send the context it were used in (the lines before it event/trigger it's in)

**Comment by MuratKandilli** [comment]
> Can you send the context it was used in (the lines before it event/trigger it's in)

the all context is 
"command /givemyuuid:
trigger:
set {_uuid} to uuid of player
broadcast "%{_uuid}%""

**Comment by bluelhf** [comment]
Can you send the full console log?

**Comment by MuratKandilli** [comment]
> Can you send the full console log?

https://pastebin.com/6WkVMv4T

**Comment by TPGamesNL** [comment]
Can you send your entire server in a zip file? Also, can you share your Java version?

**Comment by MuratKandilli** [comment]
> Can you send your entire server in a zip file? Also, can you share your Java version?

https://dosya.co/5i0ghn3j87zd/local.rar.html

java version is 17.0.3

**Comment by TPGamesNL** [comment]
I tried with that exact server, same Java version (also tried Temurin, saw you're using that on Discord) on both Windows and Ubuntu, but I can't replicate on any of them. A similar issue was reported a while ago: https://github.com/SkriptLang/Skript/issues/4712, but it got left without response.

(sent by you on Discord, for reference:)
![image](https://user-images.githubusercontent.com/29547183/174487985-b2f5dd95-3338-4f94-942f-61b36c2898f7.png)

Do you host the server on your own machine, or do you use an online server host (if so, which one)? Can you give the OS specifications as well? Can you also try with the following Skript jar (unzip first, GH doesn't like me uploading jar files): [Skript.zip](https://github.com/SkriptLang/Skript/files/8935820/Skript.zip)
 
I know these are a lot of things, but I really don't know why I'm not able to replicate it myself

**Comment by AyhamAl-Ali** [comment]
something that can be related, syntax of ExprUUID is all in captical 
https://github.com/SkriptLang/Skript/blob/0c74e07bdd9ae12cce9c6f1de716608bc72b05b3/src/main/java/ch/njol/skript/expressions/ExprUUID.java#L54

Which iirc that matters if they have `case sensitive` in config.sk set to true
https://github.com/SkriptLang/Skript/blob/0c74e07bdd9ae12cce9c6f1de716608bc72b05b3/src/main/resources/config.sk#L139

Can you double check if that value is set to true?
whether yes or not try using `player's UUID` and see if that fixes it

**Comment by TPGamesNL** [comment]
@AyhamAl-Ali indeed, the upper case is at fault. @MuratKandilli you can ignore all the questions above, I can replicate it now.

[LiteralPatternElement](https://github.com/SkriptLang/Skript/blob/0c74e07bdd9ae12cce9c6f1de716608bc72b05b3/src/main/java/ch/njol/skript/patterns/LiteralPatternElement.java#L32) uses locale-dependent lower case conversion, which gives issues when setting your language to Turkish (`-Duser.language=tr`). Fix by replacing with `toLowerCase(Locale.ENGLISH)` (similar for upper case conversion).

This also occurs in many more places in Skript (most probably unwanted, though things like ExprStringCase should be locale-dependent)
![image](https://user-images.githubusercontent.com/29547183/174489937-026cad8e-6586-4c4b-a75f-61465cd9851a.png)

We could add something about this to the code conventions document, to make sure this doesn't happen in the future.

**Comment by AyhamAl-Ali** [comment]
Well done on tracing it back @TPGamesNL 👏

**Comment by APickledWalrus** [comment]
Thank you very much for this report.

A nightly build is available with the proper fixes for this issue included. It should be the same as 2.6.2, just with some additional bugs fixed. You can download it at: https://github.com/SkriptLang/Skript/suites/6997256121/artifacts/274289708 (you must be logged into your GitHub account to download)

**Comment by MuratKandilli** [comment]
thank you all for your help  guys



> Thank you very much for this report.
> 
> A nightly build is available with the proper fixes for this issue included. It should be the same as 2.6.2, just with some additional bugs fixed. You can download it at: https://github.com/SkriptLang/Skript/suites/6997256121/artifacts/274289708 (you must be logged into your GitHub account to download)

I downloaded the .jar and fixed for me now. I will close the ıssue

**Comment by APickledWalrus** [comment]
We keep issues open until a new release containing the necessary fixes is published :)

**Comment by MuratKandilli** [comment]
> We keep issues open until a new release containing the necessary fixes is published :)

oh okay its my bad xD

**Comment by Downed-new** [comment]
https://user-images.githubusercontent.com/119142747/204114927-94253d48-e983-48aa-bccd-fedc9a139785.png

might not be the same but i have no idea why this is happening, it worked a few hours ago. i have tried with and without tuske because i dont remember if i had it installed. but for some reason some of my skript doesnt work when i have tuske installed.

**Comment by sovdeeth** [comment]
@Downed-new Tuske does not work with the latest versions of Skript. Do not use it.
Additionally, this is not a bug. You're trying to use the variables section to set a default value, but that only works if the input is a type. As the error says, `uuid of player` is not a type. You could put `text` there instead, though.

**Comment by Downed-new** [comment]
Ok thank you very much @sovdeeth, i didn't know you couldn't set it to a number like that.
